frontend/src/components/modals/editAccount/EditAccountModal.tsx

```diff
@@ -14,57 +16,69 @@ import { useOnClickOutside } from "usehooks-ts";
 import { ModalSpan } from "../GenericModalComponents.styled";
 
 const EditAccountModal = ({ hideModal }: IModal) => {
-	const { newUsername, handleNicknameChange, updateUsername } =
-		useModalEditUsername(hideModal);
+    const {
+        file,
+        fileTypes,
+        handleChange,
+        onTypeError,
+        onSizeError,
+        onDrop,
+        onSelect,
+        onDraggingStateChange,
+        submitChanges
+    } = useModalEditAvatar();
+    const {
+        handleSubmit,
+        handleChange: handleInputChange,
+        values,
+        errors,
+        touched
+    } = useModalEditUsername(hideModal, file);
 
-	const {
-		file,
-		fileTypes,
-		handleChange,
-		onTypeError,
-		onSizeError,
-		onDrop,
-		onSelect,
-		onDraggingStateChange,
-		submitChanges
-	} = useModalEditAvatar();
+    const ref = useRef(null);
 
-	const ref = useRef(null);
+    useOnClickOutside(ref, hideModal);
 
-	useOnClickOutside(ref, hideModal);
-
-	return (
-		<EditAccountWrapper ref={ref}>
-			<OptionEditAccount>
-				<ModalSpan>Twój nowy nick:</ModalSpan>
-				<TextInput
-					value={newUsername}
-					onChange={handleNicknameChange}
-				/>
-			</OptionEditAccount>
-
-			<OptionEditAvatar>
-				<ModalSpan>Nowy avatar:</ModalSpan>
-				<FileUploader
-					handleChange={handleChange}
-					name="file"
-					types={fileTypes}
-					label="Kliknij aby dodać lub upuść nowy avatar."
-					multiple={false}
-					hoverTitle="Upuść aby dodać"
-					onTypeError={onTypeError}
-					maxSize={1}
-					onSizeError={onSizeError}
-					onDrop={onDrop}
-					onSelect={onSelect}
-					onDraggingStateChange={onDraggingStateChange}
-					dropMessageStyle={{ backgroundColor: "fuchsia" }}
-				/>
-			</OptionEditAvatar>
-
-			<PrimaryButton onClick={updateUsername}>Zapisz zmiany</PrimaryButton>
-		</EditAccountWrapper>
-	);
+    return (
+        <EditAccountWrapper
+            ref={ref}
+            onSubmit={handleSubmit}>
+            <OptionEditAccount>
+                <InputWrapper>
+                    <ModalSpan>Twój nowy nick:</ModalSpan>
+                    <TextInput
+                        type="text"
+                        id="login"
+                        name="login"
+                        onChange={handleInputChange}
+                        value={values.login}
+                    />
+                </InputWrapper>
+                {errors.login && touched.login ? (
+                    <Error>{errors.login}</Error>
+                ) : null}
```

Comment:
this "null" is required, u can't do it like touche.login && <Error> ... ?

Reply:
Done